Cython/Shadow.pyi

```diff
@@ -20,8 +23,8 @@ ushort = py_int
 uchar = py_int
 size_t = py_int
 Py_ssize_t = py_int
-Py_UCS4 = Union[py_int, str]
-Py_UNICODE = Union[py_int, str]
+Py_UCS4 = py_int | str
```

Comment:
I have a feeling that this will not be accepted by older versions of the type checkers

Reply:
It won't, but type checkers aren't tied to a Python version.

Even though this syntax is from Python 3.10 (or 3.9; I don't remember), all type checkers support it as far as I know.

This works because it is a stubfile.

Reply:
That said, the tests are failing this pull request on older versions because of this. Which I think we need to change the tests for, once `Type` and `Union` is removed it will be impossible to support from Python 2 up to the latest version.